lib/CsvToMarkdown.js

```diff
@@ -20,7 +20,8 @@ function csvToMarkdown(csvContent, delimiter, hasHeader) {
         if (typeof tabularData[i] == "undefined") {
             tabularData[i] = [];
         }
-        var row = e.split(delimiter);
+        var regex = new RegExp('(?!\\B"[^"]*)' + delimiter + '(?![^"]*"\\B)');
```

Comment:
Sorry for the delay to reply! Thanks for reviewing the PR.

Yes, you're right, having the expression both sides isn't necessary. I came up with that solution to use it on a project I was working on at the moment, but hadn't had time to play with it and test new stuff. I've tried the regex101 above with other sample strings and still got the same matches.